megamek/src/megamek/common/Minefield.java

```diff
@@ -41,6 +41,9 @@
     public static final int CLEAR_NUMBER_INF_ENG_ACCIDENT = 3;
     public static final int CLEAR_NUMBER_BA_SWEEPER = 6;
     public static final int CLEAR_NUMBER_BA_SWEEPER_ACCIDENT = 2;
+    
+    public static final int BAP_DETECT_TARGET_PREPLACED = 10;
+    public static final int BAP_DETECT_TARGET_WEAPON_DELIVERED = 7;
```

Comment:
I'm completely unconvinced that allowing users to customize these target numbers and other constants is the way to go. This will lead to situations like having a client indicate that some action needs an x+ while the server thinks it's a y+.

Reply:
I've not done the design for how to handle it, but... I do know it is handled in other programs and thus is very much possible to do. Further, our current code base (outside of versioning as of #3162) also assumes the same base data setup in the majority of cases.

My initial thought process is that we have a localhost-based setup that is sent to the connected clients, but I've not taken it past that given it would be a significantly harder undertaking to find and implement instead of just implementing the change.